scripts/restream.py

```diff
@@ -71,37 +70,46 @@ def setupPlayer(twitch_name, player_num):
 
     print("Found fps", fps)
 
-    size_re = re.compile(", \\d+x(?P<size_key>(360|480|720|1080))(,| \\[SAR)")
+    sizeRegex = re.compile(", (?P<width>\\d+)x(?P<height>\\d+)(,| \\[SAR)")
 
-    m = size_re.search(str(stderr))
+    m = sizeRegex.search(str(stderr))
 
-    width, height = RESOLUTIONS[m.group("size_key")]
+    width = int(m.group("width"))
+    height = int(m.group("height"))
 
     print("Found size", width, height)
 
-    # Update player config file
-    player_config = (
-        config_base_file.replace("USER_NAME", twitch_name)
-        .replace("TWITCH_NAME", twitch_name)
-        .replace("TWITCH_URL", twitch_url)
-        .replace("STREAM_URL", local_url)
-        .replace("PORT", str(ocr_dest_port))
-        .replace("SCAN_RATE", str(fps))
-        .replace("CAP_WIDTH", str(round(width * CAP_RATIOS[0])))
-        .replace("CAP_HEIGHT", str(round(height * CAP_RATIOS[1])))
-    )  # TODO read from ffmpeg instead
-
-    player_config_filename = "config.competition.p{}.json".format(player_num)
-
-    with open(player_config_filename, "w") as content_file:
-        content_file.write(player_config)
-
-    # Run calibrator on player stream
-    p = Popen(["python3", "calibrate.py", "--config", player_config_filename])
+    # ==================================
+    # 3. write player config file from template
+
+    playerConfigFile = "config.competition.p{}.json".format(playerNum)
+
+    os.remove(playerConfigFile)
+
+    config = Config(playerConfigFile, auto_save=False, default_config=TEMPLATE_FILE)
+
+    config["player.name"] = twitchName
```

Comment:
So clever 👏 The override is a little hidden, but this is such a low cost tweak to the previous hacky version, so very nice!

Reply:
:clap::clap::clap:

Reply:
That very recent addition of `default_config` was great to use for this case! :D